fix(popup): delete EventListener

MrDeerly opened the pull request:
This MR aims to fix the popup mixin. Currently there is a bug with creating and removing the eventListeners. This leads to the error `this.triggerlEl is not a function`. This MR relies on the official uiv implementation

mtorromeo commented:
Hi, thanks for the fix.

Do you mind updating the MR to just include the fix itself without changing all the `addEventListener`/`removeEventListener` calls to `this.on`/`this.off`?

MrDeerly commented:
> Hi, thanks for the fix.
> 
> Do you mind updating the MR to just include the fix itself without changing all the `addEventListener`/`removeEventListener` calls to `this.on`/`this.off`?

lol.. somehow i managed to miss that just the missing `removeEventListener` is the issue.. I'll adjust the fix later that day by just adding the missing `removeEventListener` :+1:

mtorromeo commented on src/mixins/popupMixin.js:
```diff
@@ -294,17 +294,17 @@ export default {
+        this.triggerEl.removeEventListener('focus', this.handleAuto);
+        this.triggerEl.removeEventListener('blur', this.handleAuto);
       }
       window('click', this.windowClicked);
```
I think in your previous version you had a valid fix here also: `window` -> `window.removeEventListener`

MrDeerly commented:
@mtorromeo i removed the unnecessary helper functions and just added the missing `removeEventListener` calls + changed `beforeDestroy` to `beforeUnmount`

(still not sure how i missed that there is already the eventlistener creation :joy:
